src/Database/Console/PruneCommand.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Winter\Storm\Database\Console;
+
+use Exception;
+use Illuminate\Database\Console\PruneCommand as BasePruneCommand;
+use Illuminate\Database\Eloquent\MassPrunable;
+use Illuminate\Database\Eloquent\Prunable;
+use Winter\Storm\Support\Facades\Event;
+use Winter\Storm\Support\Facades\File;
+
+class PruneCommand extends BasePruneCommand
+{
+    /**
+     * {@inheritDoc}
+     */
+    protected function models()
+    {
+        if (! empty($models = $this->option('model'))) {
+            return collect($models)->filter(function ($model) {
+                return class_exists($model);
+            })->values();
+        }
+
+        $except = $this->option('except');
+
+        return $this->findModels()
+            ->when(! empty($except), function ($models) use ($except) {
+                return $models->reject(function ($model) use ($except) {
+                    return in_array($model, $except);
+                });
+            })->filter(function ($model) {
+                return $this->isPrunable($model);
+            })->filter(function ($model) {
+                return class_exists($model);
+            })->values();
+    }
+
+    protected function findModels()
+    {
+        /**
+         * @event system.console.model.prune.findModels
+         * Give the opportunity to return a collection of Models to prune.
+         *
+         * Example usage:
+         *
+         *     Event::listen('system.console.model.prune.findModels', function () {
+         *         return collect(['example model' => '\System\Models\File']);
+         *     });
+         *
+         */
+        $models = Event::fire('system.console.model.prune.findModels', [$this], true);
+        if ($models instanceof \Illuminate\Support\Collection) {
+            return $models;
+        }
+
+        $paths = [
+            base_path() . '/modules' => '/*/models',
+            plugins_path() => '/*/*/models',
+        ];
+        return File::findModels($paths);
+    }
+
+    /**
+     * Determine if the given model class is prunable.
+     *
+     * @param  string  $model
+     * @return bool
+     */
+    protected function isPrunable($model)
+    {
+        try {
+            $uses = class_uses_recursive($model);
+        } catch (Exception $e) {
+            return false;
+        }
+
+        return in_array(Prunable::class, $uses) || in_array(MassPrunable::class, $uses);
+    }
+}
```

Comment:
@mjauvin let's just include this in the System module directly. it's pretty dependent on that layer of Winter rather than being able to standalone in Storm.